src/Cube.cpp

```diff
@@ -87,18 +88,18 @@ std::shared_ptr<Mesh> Cube::create_cube() const
 {
     std::vector<Vertex> const vertices = m_big_cube ? InternalMeshData::big_cube.vertices : InternalMeshData::cube.vertices;
     std::vector<u32> const indices = m_big_cube ? InternalMeshData::big_cube.indices : InternalMeshData::cube.indices;
-    std::vector<Texture> textures;
+    std::vector<std::shared_ptr<Texture>> textures;
```

Comment:
Move to the shared_ptr<Texture> is asking to be a commit of its own. This would help us find bugs easier if we find any